device/accton/x86_64-accton_csp7551-r0/sonic_platform/sfp.py

```diff
@@ -0,0 +1,1753 @@
+SFP_TX_DISABLE_SOFT_BIT = 6
+
+sfp_cable_length_tup = ('LengthSMFkm-UnitsOfKm', 'LengthSMF(UnitsOf100m)',
+                        'Length50um(UnitsOf10m)', 'Length62.5um(UnitsOfm)',
+                        'LengthCable(UnitsOfm)', 'LengthOM3(UnitsOf10m)')
+
+sfp_compliance_code_tup = ('10GEthernetComplianceCode', 'InfinibandComplianceCode',
+                           'ESCONComplianceCodes', 'SONETComplianceCodes',
+                           'EthernetComplianceCodes', 'FibreChannelLinkLength',
+                           'FibreChannelTechnology', 'SFP+CableTechnology',
+                           'FibreChannelTransmissionMedia', 'FibreChannelSpeed')
+
+# definitions of the offset and width for values in XCVR_QSFP_DD info eeprom
+XCVR_EXT_TYPE_OFFSET_QSFP_DD = 72
+XCVR_EXT_TYPE_WIDTH_QSFP_DD = 2
+XCVR_CONNECTOR_OFFSET_QSFP_DD = 75
+XCVR_CONNECTOR_WIDTH_QSFP_DD = 1
+XCVR_CABLE_LENGTH_OFFSET_QSFP_DD = 74
+XCVR_CABLE_LENGTH_WIDTH_QSFP_DD = 1
+XCVR_HW_REV_OFFSET_QSFP_DD = 36
+XCVR_HW_REV_WIDTH_QSFP_DD = 2
+XCVR_VENDOR_DATE_OFFSET_QSFP_DD = 54
+XCVR_VENDOR_DATE_WIDTH_QSFP_DD = 8
+XCVR_DOM_CAPABILITY_OFFSET_QSFP_DD = 2
+XCVR_DOM_CAPABILITY_WIDTH_QSFP_DD = 1
+XCVR_MEDIA_TYPE_OFFSET_QSFP_DD = 85
+XCVR_MEDIA_TYPE_WIDTH_QSFP_DD = 1
+XCVR_FIRST_APPLICATION_LIST_OFFSET_QSFP_DD = 86
+XCVR_FIRST_APPLICATION_LIST_WIDTH_QSFP_DD = 32
+XCVR_SECOND_APPLICATION_LIST_OFFSET_QSFP_DD = 351
+XCVR_SECOND_APPLICATION_LIST_WIDTH_QSFP_DD = 28
+
+# definitions of the offset for values in QSFP_DD info eeprom
+QSFP_DD_TYPE_OFFSET = 0
+QSFP_DD_VENDOR_NAME_OFFSET = 1
+QSFP_DD_VENDOR_PN_OFFSET = 20
+QSFP_DD_VENDOR_SN_OFFSET = 38
+QSFP_DD_VENDOR_OUI_OFFSET = 17
+
+#definitions of the offset and width for values in DOM info eeprom
+QSFP_DD_TEMPE_OFFSET = 14
+QSFP_DD_TEMPE_WIDTH = 2
+QSFP_DD_VOLT_OFFSET = 16
+QSFP_DD_VOLT_WIDTH = 2
+QSFP_DD_TX_BIAS_OFFSET = 42
+QSFP_DD_TX_BIAS_WIDTH = 16
+QSFP_DD_RX_POWER_OFFSET = 58
+QSFP_DD_RX_POWER_WIDTH = 16
+QSFP_DD_TX_POWER_OFFSET = 26
+QSFP_DD_TX_POWER_WIDTH = 16
+QSFP_DD_CHANNL_MON_OFFSET = 26
+QSFP_DD_CHANNL_MON_WIDTH = 48
+QSFP_DD_CHANNL_DISABLE_STATUS_OFFSET = 86
+QSFP_DD_CHANNL_DISABLE_STATUS_WIDTH = 1
+QSFP_DD_CHANNL_RX_LOS_STATUS_OFFSET = 19
+QSFP_DD_CHANNL_RX_LOS_STATUS_WIDTH = 1
+QSFP_DD_CHANNL_TX_FAULT_STATUS_OFFSET = 7
+QSFP_DD_CHANNL_TX_FAULT_STATUS_WIDTH = 1
+QSFP_DD_MODULE_THRESHOLD_OFFSET = 0
+QSFP_DD_MODULE_THRESHOLD_WIDTH = 72
+QSFP_DD_CHANNL_STATUS_OFFSET = 26
+QSFP_DD_CHANNL_STATUS_WIDTH = 1
+
+XCVR_TYPE_WIDTH = 1
+QSFP_DD_DOM_BULK_DATA_START = 14
+QSFP_DD_DOM_BULK_DATA_SIZE = 4
+
+SFP_TYPE_CODE_LIST = [
+    '03' # SFP/SFP+/SFP28
+]
+QSFP_TYPE_CODE_LIST = [
+    '0d', # QSFP+ or later
+    '11' # QSFP28 or later
+]
+QSFP_DD_TYPE_CODE_LIST = [
+    '18', # QSFP-DD Double Density 8X Pluggable Transceiver
+    '1e'
+]
+
+QSFP_PORT_START = 1
+QSFP_PORT_END = 32
+QSFP_DD_PORT_START = 0
+QSFP_DD_PORT_END = 0
+SFP_PORT_START = 0
+SFP_PORT_END = 0
+FPGA_PORT_START=33
+FPGA_PORT_END=56
+
+class Sfp(SfpBase):
```

Comment:
Ideally, `class Sfp` should be inherited from `SfpOptoeBase`. For more details, please refer to https://github.com/Azure/SONiC/blob/master/doc/sfp-refactor/sfp-refactor.md#sfpoptoebase

Reply:
E.g., https://github.com/Azure/sonic-buildimage/blob/761ae244271d01f60bb6a4462888590066f78338/platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py#L16

Reply:
Ok, I'll modify it.